src/lib.rs

```diff
@@ -66,28 +73,158 @@ pub fn scale_alpha_to_original_coverage(
     }
     alpha_scaled_data
 }
+// Defines a line of weights. `coefficients` contains a weight for each pixel after `start`
+#[derive(Debug, Clone)]
+struct CachedWeight {
+    pub start: u32,
+    pub coefficients: Rc<Vec<f32>>,
+}
+
+pub(crate) fn calculate_weights(src: u32, target: u32, filter_scale: f32) -> Vec<CachedWeight> {
+    assert!(
+        src > target,
```

Comment:
We allow resampling in only width or height, but this assertion seems to break this principle